pkg/controller/storagecluster/reconcile.go

```diff
@@ -690,6 +747,13 @@ func (r *ReconcileStorageCluster) ensureCephCluster(sc *ocsv1.StorageCluster, re
 	}
 	objectreferencesv1.SetObjectReference(&sc.Status.RelatedObjects, *objectRef)
 
+	type mapNegativeConditionsFunc func(*[]conditionsv1.Condition, *cephv1.CephCluster)
```

Comment:
Rather than doing this up here, it seems like it would make more sense to just do an `if/else` at like 766 to pick between mapping functions.

Reply:
Reasoning was: felt much more clutter if you put a if/else inside another else and since `Go` provides more functional style programming... (=
Changed...